Doxygen: Add missing generic enums.hpp file to the documentation

AlexandraTrifan opened the pull request:
Important enums, like trigger conditions, trigger mode or trigger source are in this generic enums.hpp file and they all need to be exposed in the Doxygen docs.

Signed-off-by: Alexandra.Trifan <Alexandra.Trifan@analog.com>

commodo commented:
the commit could use a bit more description;
no strong preference from my side about it;

commodo commented:
build-status however is still failing

AlexandraTrifan commented:
Added some description to the commit. 
The Travis CI failed because some Doxygen stuff were not correctly added. I fixed them now, the PR build still fails, but the commit build looks fine. 
Can I merge it?

commodo commented:
> Can I merge it?

yes;
build has failed because of infrastructure stuff, which is not our problem
